src/dos/program_more.cpp

```diff
@@ -0,0 +1,504 @@
+
+	auto decision = Decision::Terminate;
+	while (!shutdown_requested) {
+		CALLBACK_Idle();
+
+		uint16_t count = 1;
+		char choice    = 0;
+		DOS_ReadFile(STDIN, reinterpret_cast<uint8_t *>(&choice), &count);
+
+		if (count == 0 || choice == code_ctrl_c || choice == code_esc ||
+		    choice == 'q' || choice == 'Q') {
+			decision = Decision::Terminate;
+			break;
+		}
+
+		if (choice == code_cr || choice == ' ') {
+			decision = Decision::More;
+			break;
+		}
+
+		if (!multiple_files)
+			continue;
+
+		if (choice == 'n' || choice == 'N') {
+			decision = Decision::NextFile;
+			break;
+		}
+	}
+
+	if (decision == Decision::Terminate || decision == Decision::NextFile) {
+		WriteOut(" ");
+		WriteOut(MSG_Get("SHELL_CMD_MORE_TERMINATE"));
+		WriteOut("\n");
+		++line_counter;
+	} else {
+		// We are going to continue - erase the prompt
+		WriteOut("\033[M"); // clear line
+		auto counter = GetCurrentColumn();
+		while (counter--)
+			WriteOut("\033[D"); // cursor one position back
+	}
+
+	return decision;
+}
+
+bool MORE::GetCharacter(char &code)
+{
+	if (!tabs_remaining) {
+		while (true) {
+			// Retrieve character from input stream
+			uint16_t count = 1;
+			DOS_ReadFile(input_handle,
+			             reinterpret_cast<uint8_t *>(&code),
+			             &count);
+
+			if (!count) {
+				return false; // end of stream
+			}
+
+			if (ctrl_c_enable && code == code_ctrl_c) {
+				if (input_files.empty()) {
+					WriteOut("^C");
+				}
+				return false; // quit by CTRL+C
+			}
+
+			// Skip CR/LF characters if requested
+			if (skip_next_cr && code == code_cr) {
+				skip_next_cr = false;
+			} else if (skip_next_lf && code == code_lf) {
+				skip_next_lf = false;
+			} else {
+				break;
+			}
+		}
+
+		// If TAB found, replace it with given number of spaces
+		if (code == '\t') {
+			tabs_remaining = tab_size;
+		}
+	}
+
+	if (tabs_remaining) {
+		--tabs_remaining;
+		code = ' ';
+	}
+
+	return true;
+}
+
+uint8_t MORE::GetCurrentColumn()
+{
+	const auto page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+	return CURSOR_POS_COL(page);
+}
+
+uint8_t MORE::GetCurrentRow()
+{
+	const auto page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+	return CURSOR_POS_ROW(page);
+}
+
+void MORE::AddMessages()
+{
+	MSG_Add("SHELL_CMD_MORE_HELP",
+	        "Display command output or text file one screen at a time.\n");
+	MSG_Add("SHELL_CMD_MORE_HELP_LONG",
+	        "Display command output or text file one screen at a time.\n"
+	        "\n"
+	        "Usage:\n"
+	        "  [color=cyan]COMMAND[reset] | [color=green]more[reset] [/t[color=white]n[reset]]\n"
+	        "  [color=green]more[reset] [/t[color=white]n[reset]] < [color=cyan]FILE[reset]\n"
+	        "  [color=green]more[reset] [/t[color=white]n[reset]] [color=cyan]PATTERN[reset] [[color=cyan]PATTERN[reset] ...]\n"
+	        "\n"
+	        "Where:\n"
+	        "  [color=cyan]COMMAND[reset] is the command to display output one screen at a time.\n"
```

Comment:
"to display **the** output **of**"?

Reply:
Omit "one screen at a time", it's redundant.

Reply:
OK, changed.